src/app-layer.c

```diff
@@ -1029,11 +1029,54 @@ void AppLayerListSupportedProtocols(void)
 }
 
 /***** Setup/General Registration *****/
+static void AppLayerNamesSetup(void)
+{
+    AppProtoRegisterProtoString(ALPROTO_UNKNOWN, "unknown");
```

Comment:
can these move into the parser registrations? E.g.
```c
  /**
   * \brief Register the SMTP Protocol parser.
   */
  void RegisterSMTPParsers(void)
  {
      const char *proto_name = "smtp";

     AppProtoRegisterProtoString(ALPROTO_SMTP, proto_name);
```

Reply:
I did not find how and it is not obvious : 

This does not work for rust parsers because they do not know the `ALPROTO_SSH` value and they find it using the string `"ssh"` passed to `AppLayerRegisterProtocolDetection`

Can this be studied as a next step, also making `ALPROTO_SNMP` dynamic ?

Reply:
Yup